src/app/components/send-coins-form/send-coins-form.component.ts

```diff
@@ -26,12 +26,42 @@ export class SendCoinsFormComponent implements OnInit {
   }
 
   getCurrencies() {
-    this.currenciesService.getCurrencies()
+    this.changellyApi.getCurrencies()
       .subscribe(
         currencies => {
-          // @TODO check data for correct format and display view error if wrong
-          this.currencies = currencies;
+          if(this.checkData(currencies))
+            this.currencies = currencies
+            this.isDisabled = false
         },
-        error => this.errorMessage = <any>error)
+        error => {
+          console.log('err', error)
+          this.displayError(error)
+        })
   }
+
+  displayError(error){
+    let errMsg = ''
+    switch(error.slice(0,3)){
```

Comment:
i think this type of thing should be moved to the view rather than having the text in the controller. look at doing ng-if on the error code and showing / hiding various text blocks there.